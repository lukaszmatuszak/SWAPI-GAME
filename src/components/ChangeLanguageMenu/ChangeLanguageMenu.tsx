import React, { ChangeEvent, useState } from 'react';
import {
    FormControl,
    RadioGroup,
    FormControlLabel,
    Radio,
} from '@material-ui/core';
import { useTranslation } from 'react-i18next';

const ChangeLanguageMenu: React.FC = () => {
    const { i18n } = useTranslation();
    const [selectedLng, setSelectedLng] = useState(i18n.language);

    const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
        setSelectedLng(event.target.value);
        i18n.changeLanguage(event.target.value);
    };

    return (
        <FormControl component="fieldset">
            <RadioGroup
                aria-label="position"
                name="position"
                value={selectedLng}
                onChange={handleChange}
                row
            >
                <FormControlLabel
                    value="en"
                    control={<Radio />}
                    label="English"
                    labelPlacement="start"
                />
                <FormControlLabel
                    value="pl"
                    control={<Radio />}
                    label="Polish"
                    labelPlacement="start"
                />
            </RadioGroup>
        </FormControl>
    );
};

export default ChangeLanguageMenu;
