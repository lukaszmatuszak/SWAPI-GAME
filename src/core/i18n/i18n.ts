import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';

import locales from './locales/index';

i18n.use(initReactI18next)
    .use(LanguageDetector)
    .init({
        ns: ['Common'],
        defaultNS: 'Common',
        resources: locales,
        fallbackLng: 'en',
        interpolation: {
            escapeValue: false,
        },
    });
