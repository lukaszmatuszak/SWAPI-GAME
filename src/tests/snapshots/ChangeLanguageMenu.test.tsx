import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ChangeLanguageMenu from '../../components/ChangeLanguageMenu/ChangeLanguageMenu';
import '../../core/i18n/i18n';

describe('ChangeLanguageMenu Snapshot - ', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<ChangeLanguageMenu />);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
