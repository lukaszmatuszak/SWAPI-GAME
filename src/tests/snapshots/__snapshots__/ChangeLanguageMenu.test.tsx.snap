// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeLanguageMenu Snapshot -  renders correctly 1`] = `
<WithStyles(ForwardRef(FormControl))
  component="fieldset"
>
  <ForwardRef(RadioGroup)
    aria-label="position"
    name="position"
    onChange={[Function]}
    row={true}
    value="en"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={<ForwardRef(WithStyles) />}
      label="English"
      labelPlacement="start"
      value="en"
    />
    <WithStyles(ForwardRef(FormControlLabel))
      control={<ForwardRef(WithStyles) />}
      label="Polish"
      labelPlacement="start"
      value="pl"
    />
  </ForwardRef(RadioGroup)>
</WithStyles(ForwardRef(FormControl))>
`;
